cmd/kubeadm/app/preflight/checks.go

```diff
@@ -415,14 +415,18 @@ func (hst HTTPProxyCheck) Name() string {
 // Check validates http connectivity type, direct or via proxy.
 func (hst HTTPProxyCheck) Check() (warnings, errors []error) {
 
-	url := fmt.Sprintf("%s://%s:%d", hst.Proto, hst.Host, hst.Port)
+	urlFmt := "%s://%s:%d"
+	if strings.Contains(hst.Host, ":") {
```

Comment:
is this a fullproof way to check this URL is IPv6? in other PRs we did something like this:

https://github.com/kubernetes/kubernetes/pull/53484/files#diff-86d5c7040c142720e7e2c9894044c353R227

maybe we could centralise as helper functions

Reply:
Don't build URLs with string formatting. Use net.JoinHostPort, and (&url.URL{Scheme:…, Host:…}).String()

Reply:
+1 to @liggitt 's suggestion, another example is here - https://github.com/kubernetes/kubernetes/pull/57846/commits/c258d4df84089b08b9cbd37b1dee4b00576a2532

Reply:
Thanks, will update.